proposed/http-handlers/request-handlers.md

```diff
@@ -110,20 +110,18 @@ use Psr\Http\Message\ResponseInterface;
 use Psr\Http\Message\ServerRequestInterface;
 
 /**
+ * Participant in processing a server request and response
+ *
  * An HTTP middleware component participates in processing an HTTP message,
- * either by acting on the request or the response. This interface defines the
- * methods required to use the middleware.
+ * either by acting on the request or the response. The middleware may create a
```

Comment:
"or the response" should become "or the response returned by the provided request handler".

Reply:
Would that be accurate though? Considering that the response *might* be generated by the middleware itself?

Reply:
Suggested: "either by acting on the request, generating the response, or forwarding the request to a subsequent middleware and possibly acting on its response."

Reply:
I like your suggestion @Crell, it doesn't close any door and it's explicit.

Do we need to explicitly say that a middleware can also simply have side effects and not do anything else (well except forwarding the request)? (e.g. write to a log)

Or we could also simply say:

> An HTTP middleware takes an HTTP request and returns an HTTP response. The middleware can delegate generating the response to a request handler.

What a middleware does and how it works is an implementation detail (and the world of middlewares out there is really diverse, which is its strength). We don't want to accidentally narrow it.

> if conditions prevent the request from being processed further.

This is IMO a problem, some middlewares may return a response just because it does, e.g. *if* you write the router as a middleware (and assuming it acts as a dispatcher as well) it's fine that it returns a response. Same if you write controllers as middlewares, or if you integrate several applications together into a pipe or router (e.g. you put a Slim and a Zend Expressive app in a router or a pipe). Not the classic use cases but still they should be allowed since these extremes are what make the middlewares so powerful.

Reply:
This change has been made.